librosa/core/spectrum.py

```diff
@@ -1281,6 +1282,152 @@ def fmt(y, t_min=0.5, n_fmt=None, kind='cubic', beta=0.5, over_sample=1, axis=-1
     return result[idx] * np.sqrt(n) / n_fmt
 
 
+@cache(level=30)
+def pcen(S, sr=22050, hop_length=512, gain=0.98, bias=2, power=0.5,
+         time_constant=0.395, eps=1e-6, b=None, max_size=1):
+    '''Per-channel energy normalization (PCEN) [1]_
+
+    This function normalizes a time-frequency representation `S` by
+    performing automatic gain control, followed by non-linear compression:
+
+        P = (S / (eps + M)**gain + bias)**power - bias**power
```

Comment:
`P[f, t] = (S[f, t] / (eps + M[f, t])**gain + bias)**power - bias**power`

Reply:
I think it's clear from context that these are element-wise operations?

Reply:
yes but laying `S[f,t]` and `M[f,t]` in full highlights which variables are arrays and which are scalars.